### PR TITLE
[#26] Implement a servant endpoint

### PR DESCRIPTION
**Pull request by rashadg1030**
Added a single endpoint for getting issues by id. Seed the DB with `sql/seed.sql` (one example issue), execute the application, and then use this endpoint: http://localhost:8081/issues/1 
This should return an issue with issueId 1.

### CODE REVIEW

**Comment by rashadg1030** [comment]
Resolved conflicts

File: src/IW/Core/Issue.hs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+data Issue = Issue
+    { issueId     :: Int
+    , issueNumber :: Int
+    , issueTitle  :: Text
+    , issueBody   :: Text
+    , issueUrl    :: Text
+    , issueRepoId :: Int
+    } deriving stock (Generic, Show, Eq)
+      deriving anyclass (FromRow)
+      deriving (FromJSON, ToJSON)
```
Comment (vrom911) [inline src/IW/Core/Issue.hs]: These two can also be derived as `anyclass`, so two lines can be merged together.

File: src/IW/Server/Types.hs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+       ) where
+
+import Servant.API.Generic (ToServantApi)
+import Servant.Server.Generic (AsServerT)
+
+import IW.App (App)
+
+
+type AppServer = AsServerT App
+type ToApi (site :: * -> *) = ToServantApi site
```
Comment (chshersh) [inline src/IW/Server/Types.hs]: Let's write here `Type -> Type` as well instead of `* -> *` for consistency.

File: src/IW/Server/Issue.hs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+issueServer :: IssueSite AppServer
+issueServer = IssueSite
+    { issueByIdRoute  = issueByIdHandler
+    }
+
+issueByIdHandler
+    :: ( WithDb env m
+       , WithError m
+       )
+    => Int
```
Comment (chshersh) [inline src/IW/Server/Issue.hs]: When we bring `Id` from `three-layer`, this argument should be changed to `Id Issue`.

File: src/IW.hs [hunk 4]
```diff
@@ -1,2 +1,32 @@
...
+mkAppEnv Config{..} = do
+    -- IO configuration
+    envDbPool <- initialisePool cDbCredentials
+
+    -- pure configuration
+    let envLogAction = mainLogAction cLogSeverity
+    pure Env{..}
+
+runServer :: AppEnv -> IO ()
+runServer env@Env{..} = run 8081 application
```
Comment (chshersh) [inline src/IW.hs]: This is just a convention, but for running backend we are usually using port `8080`. `8081` for additional server

File: src/IW/Core/Issue.hs (hunk 1 above)
Comment (rashadg1030) [reply src/IW/Core/Issue.hs]: Got it.

File: src/IW/Server/Types.hs (hunk 2 above)
Comment (rashadg1030) [reply src/IW/Server/Types.hs]: Oh yeah got it.

File: src/IW/Server/Issue.hs (hunk 3 above)
Comment (rashadg1030) [reply src/IW/Server/Issue.hs]: Got it.

File: src/IW.hs (hunk 4 above)
Comment (rashadg1030) [reply src/IW.hs]: Got it. Changing it now.

**Comment by rashadg1030** [comment]
@chshersh @vrom911 I made the fixes you suggested! I think I'm going to open an issue for introducing the `Id` type from `three-layer`.

**Pull request merged**